catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/security/app/UserApplication.java

```diff
@@ -164,6 +200,44 @@ private Map getSubjectPreferences(Subject subject) {
     return Collections.emptyMap();
   }
 
+  private List getSubjectNotifications(Subject subject) {
+    String userid = subjectIdentity.getUniqueIdentifier(subject);
+    try {
+      String filter = String.format("user = '%s'", userid);
+      List<Map<String, Object>> notificationsList =
+          persistentStore.get(PersistenceType.NOTIFICATION_TYPE.toString(), filter, 0, 999);
```

Comment:
❓ Is it safe to assume that a final page of `999` will always be accurate?

Reply:
The places where MAX_PAGE_SIZE are defined either have artifact sizes too large to embed, or I'm unable to import them - I could do something like `System.getProperty("catalog.maxPageSize")`

Reply:
Maybe @troymohl / @lessarderic / @pklinef have ideas or thoughts on this?

Reply:
That's how its gotten here: https://github.com/codice/ddf/blob/11686a8475dfa78cdd8567dfb6d338f02ebfd671/catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/QueryOperations.java#L138

Reply:
There are good reasons to limit the number of results that can come back from a single page for the catalog framework.

https://github.com/codice/ddf/blob/11686a8475dfa78cdd8567dfb6d338f02ebfd671/catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/QueryOperations.java#L98-L104

The `PersistentStoreImpl` also has a max page size limit of 1000.

https://github.com/codice/ddf/blob/master/platform/persistence/platform-persistence-core-impl/src/main/java/org/codice/ddf/persistence/internal/PersistentStoreImpl.java#L65

I think notifications sent to the front end should be limited in some way as well.  Either with a hard limit and a warning to the user that there are too many notifications to show.  Or allow paging through notifications.

Reply:
Ah, it's an upper bound for a particular feature that does not yet support paging. Makes sense. Just wanted to understand what the magic constant was doing. 

I think pulling the value from the system property will be fine.

Reply:
We can page with the `PersistentStore` so we can support it.